testing/scan.cu

```diff
@@ -640,11 +640,31 @@ void TestExclusiveScanWithBigIndexesHelper(int magnitude)
 
 void TestExclusiveScanWithBigIndexes()
 {
-  TestExclusiveScanWithBigIndexesHelper(30);
-  TestExclusiveScanWithBigIndexesHelper(31);
-  TestExclusiveScanWithBigIndexesHelper(32);
-  TestExclusiveScanWithBigIndexesHelper(33);
+    TestExclusiveScanWithBigIndexesHelper(30);
+    TestExclusiveScanWithBigIndexesHelper(31);
+    TestExclusiveScanWithBigIndexesHelper(32);
+    TestExclusiveScanWithBigIndexesHelper(33);
 }
 
 DECLARE_UNITTEST(TestExclusiveScanWithBigIndexes);
 
+struct Int {
+    int i{};
+    __host__ __device__ explicit Int(int num) : i(num) {}
+    __host__ __device__ Int() = default;
+    __host__ __device__ Int operator+(Int const& o) const { return Int{this->i + o.i}; }
+};
+
+void TestInclusiveScanWithUserDefinedType()
+{
+    thrust::device_vector<Int> vec(5, Int{1});
+
+    thrust::inclusive_scan(
+        thrust::device,
+        vec.cbegin(),
+        vec.cend(),
+        vec.begin());
+
+    ASSERT_EQUAL(static_cast<Int>(vec.back()).i, 15);
```

Comment:
I feel like I'm missing something here, but shouldn't the result be 5, not 15?

Reply:
Yes! I originally had an `std::iota` in the test but decided to remove it! Forgot that change the 15 to 5. Fixed!